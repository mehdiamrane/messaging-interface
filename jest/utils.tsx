import React from 'react';
import { render, type RenderOptions, type RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from 'src/styles/theme';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const AllProviders = ({ children }: { children: React.ReactNode }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

interface Render {
  (ui: React.ReactElement, options?: RenderOptions): RenderResult;
}

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: (props) => <AllProviders {...props} />,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render };
