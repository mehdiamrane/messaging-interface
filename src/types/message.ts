export interface Message {
  id: number;
  conversationId: number;
  authorId: number;
  timestamp: number;
  body: string;
}

export interface MessageSent {
  conversationId: number;
  authorId: number;
  timestamp: number;
  body: string;
}
