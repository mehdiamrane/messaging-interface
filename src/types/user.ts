export interface User {
  id: number;
  nickname: string;
  token: string;
}

export type BareUser = Omit<User, 'token'>;
