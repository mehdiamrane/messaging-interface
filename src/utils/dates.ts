import formatDistanceToNow from 'date-fns/formatDistanceToNow';
import intlFormat from 'date-fns/intlFormat';
import { enUS, fr } from 'date-fns/locale';

const localeMapping = { 'en-US': enUS, fr };

export const formatRelativeDate = (date: Date, locale: string): string =>
  formatDistanceToNow(date, { locale: localeMapping[locale], addSuffix: true });

export const formatDate = (date: Date, locale: string): string =>
  intlFormat(
    date,
    {
      weekday: 'long',
      year: 'numeric',
      month: 'long',
      day: 'numeric',
      hour: 'numeric',
      minute: 'numeric',
    },
    {
      locale,
    },
  );

export const timestampToDate = (timestamp: number): Date => new Date(timestamp * 1000);
