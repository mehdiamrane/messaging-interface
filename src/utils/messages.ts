import type { Message } from 'src/types/message';
import type { ChatBubbleProps } from 'src/components/conversation/chatBubble/ChatBubble';

export const sortMessages = (messages: Message[]): Message[] =>
  (messages || []).sort((a, b) => a.timestamp - b.timestamp);

export const getMessagePosition = (messages: Message[], index: number): ChatBubbleProps['position'] => {
  const previous = messages[index - 1];
  const current = messages[index];
  const next = messages[index + 1];
  if (previous?.authorId !== current.authorId && next?.authorId !== current.authorId) {
    return 'single';
  }
  if (previous?.authorId !== current.authorId && next?.authorId === current.authorId) {
    return 'first';
  }
  if (previous?.authorId === current.authorId && next?.authorId !== current.authorId) {
    return 'last';
  }
  return 'middle';
};
