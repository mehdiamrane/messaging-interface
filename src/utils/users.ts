import type { User } from 'src/types/user';

// Default way to use a logged user
export const getLoggedUserId = (): User['id'] => 1;

export const isLoggedUser = (userId: User['id']): boolean => userId === getLoggedUserId();

export const getUserData = (userId: User['id'], users: User[]): User | undefined =>
  users.find(({ id }) => id === userId);

export const getAvatarFromNickname = (nickname: User['nickname']): string =>
  `https://ui-avatars.com/api/?bold=true&size=54&background=CBD5E0&rounded=true&name=${nickname}`;
