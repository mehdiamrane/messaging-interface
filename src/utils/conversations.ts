import type { Conversation } from 'src/types/conversation';
import type { BareUser } from 'src/types/user';
import { getLoggedUserId } from './users';

export const sortConversationsByLastActive = (conversations: Conversation[]): Conversation[] =>
  conversations.sort((a, b) => b.lastMessageTimestamp - a.lastMessageTimestamp);

export const getOtherUserFromConversation = (conversation: Conversation): BareUser =>
  conversation.senderId === getLoggedUserId()
    ? { id: conversation.recipientId, nickname: conversation.recipientNickname }
    : { id: conversation.senderId, nickname: conversation.senderNickname };
