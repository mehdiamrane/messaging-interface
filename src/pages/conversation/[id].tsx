import React, { useState, useEffect, useCallback, useRef } from 'react';
import type { FC } from 'react';
import Head from 'next/head';
import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
import { GetServerSideProps } from 'next';

import dynamic from 'next/dynamic';

const ErrorBanner = dynamic(() => import('src/components/layout/errorBanner/ErrorBanner'), { ssr: false });

import { Conversation } from 'src/types/conversation';
import { Message } from 'src/types/message';

import { useTranslation } from 'next-i18next';
import Aside from 'src/components/layout/aside/Aside';
import Main from 'src/components/layout/main/Main';
// import ErrorBanner from 'src/components/layout/errorBanner/ErrorBanner';
import List from 'src/components/conversation/list/List';
import MessagesBox from 'src/components/conversation/messagesBox/MessagesBox';

import { getMessagesByConversationId, getConversationsByUserId } from 'src/services/api';
import { getLoggedUserId } from 'src/utils/users';
import { getOtherUserFromConversation } from 'src/utils/conversations';
import { BareUser } from 'src/types/user';

interface ConversationProps {
  conversations: Conversation[];
  currentConversation: Conversation;
  otherUser: BareUser;
}

const ConversationPage: FC<ConversationProps> = ({ conversations, currentConversation, otherUser }) => {
  const { t } = useTranslation('common');
  const [messages, setMessages] = useState<Message[]>();
  const [loading, setLoading] = useState<boolean>(true);
  const [error, setError] = useState<string>();

  const messagesEndRef = useRef(null);

  useEffect(() => {
    return () => {
      console.log('unmount');
      setLoading(true);
    };
  }, [currentConversation]);

  const getMessages = useCallback(async () => {
    console.log('getting messages');
    const { data: messagesData } = await getMessagesByConversationId(currentConversation.id);
    setMessages(messagesData);
    setLoading(false);
  }, [currentConversation]);

  const scrollToBottom = useCallback(() => {
    messagesEndRef.current.scrollIntoView({ behavior: 'smooth' });
  }, []);

  useEffect(() => {
    getMessages();
  }, [getMessages]);

  useEffect(() => {
    scrollToBottom();
  }, [messages, scrollToBottom]);

  const handleError = useCallback((error?: string) => {
    setError(error);
  }, []);

  return (
    <>
      <Head>
        <title>{t('conversation.meta.title', { name: otherUser.nickname })}</title>
        <meta name='description' content={t('conversation.meta.description', { name: otherUser.nickname })}></meta>
      </Head>
      <ErrorBanner error={error} onClose={() => setError(undefined)} />
      <Aside hideOnMobile>
        <List conversations={conversations} />
      </Aside>
      <Main hideOnMobile={false}>
        <MessagesBox
          fetchMessages={getMessages}
          handleError={handleError}
          loading={loading}
          messages={messages}
          conversation={currentConversation}
          messagesEndRef={messagesEndRef}
        />
      </Main>
    </>
  );
};

export default ConversationPage;

export const getServerSideProps: GetServerSideProps = async ({ params, locale }) => {
  const { id } = params;

  const { data: conversations } = await getConversationsByUserId(getLoggedUserId());

  const currentConversation = conversations.find((conv) => conv.id === Number(id));

  const otherUser = getOtherUserFromConversation(currentConversation);

  return {
    props: {
      otherUser,
      conversations,
      currentConversation,
      ...(await serverSideTranslations(locale, ['common'])),
    },
  };
};
