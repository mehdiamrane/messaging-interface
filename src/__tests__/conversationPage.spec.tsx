import React from 'react';
import { cleanup, screen, waitFor } from '@testing-library/react';
import { render } from 'jest/utils';
import ConversationPage from 'src/pages/conversation/[id]';
import { Conversation } from 'src/types/conversation';
import { Message } from 'src/types/message';
import { BareUser } from 'src/types/user';
import 'jest-styled-components';

const useRouter = jest.spyOn(require('next/router'), 'useRouter');

const messages: Message[] = [
  {
    id: 1,
    conversationId: 1,
    timestamp: 1625637849,
    authorId: 1,
    body: "Bonjour c'est le premier message de la première conversation",
  },
  {
    id: 2,
    conversationId: 1,
    timestamp: 1625637867,
    authorId: 1,
    body: "Bonjour c'est le second message de la première conversation",
  },
];

jest.mock('src/services/api', () => ({
  getMessagesByConversationId: () =>
    new Promise((resolve) => {
      resolve({ data: messages });
    }),
}));

jest.mock(
  'next/link',
  () =>
    ({ children }) =>
      children,
);

afterAll(cleanup);

describe('ConversationPage', () => {
  it('should render correctly the Conversation Page', async () => {
    useRouter.mockImplementation(() => ({
      basePath: '',
      pathname: '/',
      route: '/',
      asPath: '/',
      query: {},
      push: jest.fn(),
      replace: jest.fn(),
      reload: jest.fn(),
      back: jest.fn(),
      prefetch: jest.fn(),
      beforePopState: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
        emit: jest.fn(),
      },
      isFallback: false,
    }));

    // for messagesEndRef.current.scrollIntoView
    Element.prototype.scrollIntoView = jest.fn();

    const dummyConversations: Conversation[] = [
      {
        id: 1,
        recipientId: 2,
        recipientNickname: 'Jeremie',
        senderId: 1,
        senderNickname: 'Thibaut',
        lastMessageTimestamp: 1625637849,
      },
      {
        id: 3,
        recipientId: 1,
        recipientNickname: 'Thibaut',
        senderId: 4,
        senderNickname: 'Elodie',
        lastMessageTimestamp: 1625648667,
      },
    ];

    const dummyCurrentConversation: Conversation = {
      id: 3,
      recipientId: 1,
      recipientNickname: 'Thibaut',
      senderId: 4,
      senderNickname: 'Elodie',
      lastMessageTimestamp: 1625648667,
    };

    const dummyOtherUser: BareUser = {
      id: 4,
      nickname: 'Elodie',
    };

    const { container } = render(
      <ConversationPage
        conversations={dummyConversations}
        currentConversation={dummyCurrentConversation}
        otherUser={dummyOtherUser}
      />,
    );

    expect(screen.queryByText('Jeremie')).toBeInTheDocument();

    await waitFor(() => {
      expect(screen.queryByText("Bonjour c'est le premier message de la première conversation")).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(screen.queryByText("Bonjour c'est le second message de la première conversation")).toBeInTheDocument();
    });
    expect(container).toMatchSnapshot();
  });
});
