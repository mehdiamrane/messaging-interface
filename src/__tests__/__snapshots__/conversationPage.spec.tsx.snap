// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationPage should render correctly the Conversation Page 1`] = `
.c0 {
  display: none;
}

.c13 {
  display: initial;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #E2E8F0;
  border: 1px solid #CBD5E0;
  border-top-left-radius: 0.5rem;
  border-top-right-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 4rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c4 {
  background-color: #E2E8F0;
  border: 1px solid #CBD5E0;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  overflow: auto;
  width: 100%;
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #E2E8F0;
  border: 1px solid #CBD5E0;
  border-bottom-left-radius: 0.5rem;
  border-bottom-right-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 4rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c3 span {
  color: #1A202C;
  font-size: 1.125rem;
  font-weight: 600;
}

.c12 {
  background-color: #bee3f8;
  cursor: pointer;
  -webkit-transition: border 150ms ease,background-color 150ms ease;
  transition: border 150ms ease,background-color 150ms ease;
}

.c12:hover {
  background-color: #90cdf4;
  border: 1px solid #63b3ed;
}

.c12:active {
  background-color: #bee3f8;
  border: 1px solid #90cdf4;
}

.c12 span {
  color: #1A365D;
  font-size: 1.125rem;
  font-weight: 600;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  color: #4A5568;
  font-size: 1rem;
  font-weight: 500;
  -webkit-transition: color 150ms ease;
  transition: color 150ms ease;
}

.c10 {
  color: #718096;
  font-size: 0.875rem;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #EDF2F7;
  border-bottom: 1px solid #CBD5E0;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5rem;
  gap: 1rem;
  padding: 0 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 150ms ease;
  transition: background-color 150ms ease;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5:hover {
  background-color: #bee3f8;
}

.c5:hover .c7,
.c5:hover .c9 {
  color: #2b6cb0;
}

.c5:active {
  background-color: #ebf8ff;
}

.c5:active .c7,
.c5:active .c9 {
  color: #4299e1;
}

.c14 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.25rem;
  padding: 0 1.25rem;
  position: relative;
}

.c16 {
  color: #1A202C;
  font-size: 1.125rem;
  font-weight: 600;
}

.c15 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #bee3f8;
  border: 1px solid #90cdf4;
  border-radius: 9999px;
  color: #2a4365;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0.75rem;
  left: 0.75rem;
  width: 2.5rem;
  -webkit-transition: background-color 150ms ease,color 150ms ease;
  transition: background-color 150ms ease,color 150ms ease;
}

.c15:hover {
  background-color: #90cdf4;
  border: 1px solid #63b3ed;
  color: #1A365D;
}

.c15:active {
  background-color: #bee3f8;
  border: 1px solid #90cdf4;
  color: #2c5282;
}

.c17 {
  color: #718096;
  font-size: 1rem;
  font-weight: 500;
}

.c18 {
  background-color: #E2E8F0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem;
}

.c19 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 2rem;
}

.c22 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 2rem;
  margin-top: 0.25rem;
}

.c24 {
  color: #718096;
  font-size: 0.875rem;
  font-weight: 400;
  margin-top: 0.5rem;
}

.c25 {
  min-height: 4rem;
  overflow: hidden;
}

.c26 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.75rem;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem;
  width: 100%;
}

.c27 {
  border: 2px solid #CBD5E0;
  border-radius: 0.375rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  line-height: 1.5;
  font-size: 1rem;
  padding: 0.5rem;
  resize: none;
  -webkit-transition: border 150ms ease;
  transition: border 150ms ease;
}

.c27:hover {
  border: 2px solid #A0AEC0;
}

.c27:focus {
  border: 2px solid #63b3ed;
  outline: none;
}

.c28 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #90cdf4;
  border: 2px solid #63b3ed;
  border-radius: 0.375rem;
  color: #1A365D;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  font-weight: 600;
  padding: 0 0.75rem;
  -webkit-transition: opacity 150ms ease,border 150ms ease,background-color 150ms ease;
  transition: opacity 150ms ease,border 150ms ease,background-color 150ms ease;
}

.c28:hover {
  background-color: #63b3ed;
  border: 2px solid #4299e1;
}

.c28:active {
  background-color: #bee3f8;
  border: 2px solid #90cdf4;
}

.c28[aria-disabled='true'] {
  opacity: 0.5;
  cursor: not-allowed;
}

.c28[aria-disabled='true']:active {
  pointer-events: none;
}

.c29 {
  display: none;
}

.c20 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4299e1;
  border-top-left-radius: 1rem;
  border-top-right-radius: 1rem;
  border-bottom-right-radius: 0.375rem;
  border-bottom-left-radius: 1rem;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.75rem 1rem;
  line-height: 1.25;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 28rem;
}

.c23 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4299e1;
  border-top-left-radius: 1rem;
  border-top-right-radius: 0.375rem;
  border-bottom-right-radius: 1rem;
  border-bottom-left-radius: 1rem;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.75rem 1rem;
  line-height: 1.25;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 28rem;
}

.c21 {
  word-break: break-word;
}

@media (min-width:48em) {
  .c0 {
    display: initial;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    width: 16rem;
  }
}

@media (min-width:62em) {
  .c0 {
    width: 20rem;
  }
}

@media (min-width:48em) {
  .c13 {
    display: initial;
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    width: 100%;
  }
}

@media (min-width:48em) {
  .c14 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:48em) {
  .c15 {
    display: none;
  }
}

@media (min-width:48em) {
  .c29 {
    display: initial;
    margin-right: 0.5rem;
  }
}

<div>
  <aside
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
      >
        <span>
          list.header.text
        </span>
      </div>
      <div
        class="c4"
      >
        <a
          class="c5"
          title="list.item.title"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2754%27%20height=%2754%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="list.item.avatar.alt"
              data-nimg="intrinsic"
              decoding="async"
              src="/_next/image?url=https%3A%2F%2Fui-avatars.com%2Fapi%2F%3Fbold%3Dtrue%26size%3D54%26background%3DCBD5E0%26rounded%3Dtrue%26name%3DElodie&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fui-avatars.com%2Fapi%2F%3Fbold%3Dtrue%26size%3D54%26background%3DCBD5E0%26rounded%3Dtrue%26name%3DElodie&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fui-avatars.com%2Fapi%2F%3Fbold%3Dtrue%26size%3D54%26background%3DCBD5E0%26rounded%3Dtrue%26name%3DElodie&w=128&q=75 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="c6"
          >
            <span
              class="c7 c8"
            >
              Elodie
            </span>
            <span
              class="c7 c9 c8 c10"
            >
              10 months ago
            </span>
          </div>
        </a>
        <a
          class="c5"
          title="list.item.title"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2754%27%20height=%2754%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="list.item.avatar.alt"
              data-nimg="intrinsic"
              decoding="async"
              src="/_next/image?url=https%3A%2F%2Fui-avatars.com%2Fapi%2F%3Fbold%3Dtrue%26size%3D54%26background%3DCBD5E0%26rounded%3Dtrue%26name%3DJeremie&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fui-avatars.com%2Fapi%2F%3Fbold%3Dtrue%26size%3D54%26background%3DCBD5E0%26rounded%3Dtrue%26name%3DJeremie&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fui-avatars.com%2Fapi%2F%3Fbold%3Dtrue%26size%3D54%26background%3DCBD5E0%26rounded%3Dtrue%26name%3DJeremie&w=128&q=75 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="c6"
          >
            <span
              class="c7 c8"
            >
              Jeremie
            </span>
            <span
              class="c7 c9 c8 c10"
            >
              10 months ago
            </span>
          </div>
        </a>
      </div>
      <a
        class="c11 c12"
        title="list.footer.title"
      >
        <span>
          list.footer.text
        </span>
      </a>
    </div>
  </aside>
  <main
    class="c13"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c14"
      >
        <a
          class="c15"
          title="conversation.header.backButton.title"
        >
          <svg
            fill="none"
            height="1.75rem"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1.75rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 19l-7-7m0 0l7-7m-7 7h18"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </a>
        <span
          class="c16"
        >
          Elodie
        </span>
        <span
          class="c17"
          title="conversation.header.lastMessageTimestamp.title"
        >
          conversation.header.lastMessageTimestamp.text
        </span>
      </div>
      <div
        class="c4 c18"
      >
        <div
          class="c19"
          title="Wednesday, July 7, 2021, 8:04 AM"
        >
          <div
            class="c20"
            data-testid="chat-bubble"
          >
            <span
              class="c21"
            >
              Bonjour c'est le premier message de la première conversation
            </span>
          </div>
        </div>
        <div
          class="c22"
          title="Wednesday, July 7, 2021, 8:04 AM"
        >
          <div
            class="c23"
            data-testid="chat-bubble"
          >
            <span
              class="c21"
            >
              Bonjour c'est le second message de la première conversation
            </span>
          </div>
          <span
            class="c24"
          >
            Wednesday, July 7, 2021, 8:04 AM
          </span>
        </div>
        <div />
      </div>
      <div
        class="c11 c25"
      >
        <form
          class="c26"
        >
          <input
            class="c27"
            data-testid="messageform-input"
            minlength="1"
            placeholder="conversation.form.input.placeholder"
            required=""
            value=""
          />
          <button
            class="c28"
            data-testid="messageform-submit"
            role="button"
            title="conversation.form.button.title"
            type="submit"
          >
            <span
              class="c29"
            >
              conversation.form.button.label
            </span>
            <svg
              fill="currentColor"
              height="1.25rem"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.894 2.553a1 1 0 00-1.788 0l-7 14a1 1 0 001.169 1.409l5-1.429A1 1 0 009 15.571V11a1 1 0 112 0v4.571a1 1 0 00.725.962l5 1.428a1 1 0 001.17-1.408l-7-14z"
              />
            </svg>
          </button>
        </form>
      </div>
    </div>
  </main>
</div>
`;
